include/powerloader/download_target.hpp

```diff
@@ -180,7 +180,7 @@ namespace powerloader
 
         progress_callback_t set_progress_callback(progress_callback_t callback)
         {
-            std::exchange(callback, m_progress_callback);
+            std::exchange(m_progress_callback, callback);
```

Comment:
I didn't realize `std::exchange` was working like this, good to know.

I think this should be:
```
callback = std::exchange(m_progress_callback, callback);
```

so that the return value is the previous callback (and therefore can be kept somewhere and re-set later.).

Reply:
or
```
return std::exchange(m_progress_callback, callback);
```